### PR TITLE
Escrow Proposal Creation Flow

### PR DESCRIPTION
**Pull request by wtfsayo**
Implements #6

### CODE REVIEW

File: apps/web/.env [hunk 1]
```diff
@@ -1,6 +0,0 @@
-# DO NOT ADD SECRETS TO THIS FILE. This is a good place for defaults.
-# If you want to add secrets use `.env.local` instead.
-
```
Comment (dan13ram) [inline apps/web/.env:3]: they probably don't want you to delete this file. Instead use `.env.local` as they suggested. Next will pick up both files and combine them. That way you are changing less files also (can revert .gitignore as well)

File: apps/web/src/components/Fields/SmartInput.tsx [hunk 2]
```diff
@@ -146,6 +136,18 @@ const SmartInput: React.FC<SmartInputProps> = ({
             : undefined
         }
       />
+
```
Comment (dan13ram) [inline apps/web/src/components/Fields/SmartInput.tsx:139]: I think we should leave the error message where it was before unless you have a solid reason for moving it

File: apps/web/src/modules/create-proposal/constants/transactionType.tsx [hunk 3]
```diff
@@ -40,6 +41,12 @@ export const TRANSACTION_TYPES = {
     icon: 'airdrop',
     iconBackdrop: 'rgba(28, 182, 135, 0.1)',
   },
+  [TransactionType.ESCROW]: {
+    title: 'Escrow Milestones',
```
Comment (dan13ram) [inline apps/web/src/modules/create-proposal/constants/transactionType.tsx:45]: Is "Escrow Milestones" the agreed upon name for this ? It is fine. But maybe we can do better?

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowUtils.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
+import { Address, encodeAbiParameters, toBytes, toHex } from 'viem'
+import { create } from 'zustand'
+import { createJSONStorage, persist } from 'zustand/middleware'
+
+import { EscrowFormState, EscrowFormValues } from './EscrowForm.schema'
+
+const KLEROS_ARBITRATION_PROVIDER =
```
Comment (dan13ram) [inline apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowUtils.ts:7]: do you think we should pick up this address from `@smartinvoicexyz/*` packages? The rest looks great

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/Escrow.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  const removeTransactions = useProposalStore((state) => state.removeAllTransactions)
+
+  const { data } = useSWR<ProposalsResponse>(
+    isReady ? [SWR_KEYS.PROPOSALS, chainId, query.token, '0'] : null,
+    (_, chainId, token, page) => getProposals(chainId, token, 1, Number(0))
+  )
+
+  const lastProposalId = data?.proposals?.[0]?.proposalNumber ?? 0
+
+  const handleEscrowTransaction = async (values: EscrowFormValues) => {
```
Comment (dan13ram) [inline apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/Escrow.tsx]: could we add this into a useCallback?

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+export interface EscrowFormProps {
+  onSubmit: (values: EscrowFormValues, actions: FormikHelpers<EscrowFormValues>) => void
+  isFormSubmitting: boolean
+  escrowFormDataIpfsCID: string
+}
+
+export const MilestoneSchema = yup.object({
+  amount: yup
+    .number()
+    .min(0.1, 'Amount must be greater than 0.1 ETH')
```
Comment (dan13ram) [inline apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts]: I think we can make this minimum of 0.

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  mediaFileName: yup.string(),
+  description: yup.string(),
+})
+
+export const EscrowFormSchema = yup.object({
+  clientAddress: addressValidationSchema,
+  recipientAddress: addressValidationSchema,
+  safetyValveDate: yup
+    .date()
+    .required('Safety valve date must be at least 30 days from today.'),
```
Comment (dan13ram) [inline apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts]: Could we make a check to ensure its atleast 30 days after the enddate of the last milestone ?

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.tsx [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+import { useEscrowFormStore } from './EscrowUtils'
+
+const MilestoneForm: React.FC<{
+  index: number
+  setIsMediaUploading: React.Dispatch<React.SetStateAction<boolean>>
+  removeMilestone: () => void
+}> = ({ index, removeMilestone, setIsMediaUploading }) => {
+  const formik = useFormikContext<EscrowFormValues>()
+
+  const handleRemoveMilestone = useCallback(() => {
```
Comment (dan13ram) [inline apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.tsx]: redundant useCallback.

File: apps/web/src/components/Fields/SmartInput.tsx (hunk 2 above)
Comment (wtfsayo) [reply apps/web/src/components/Fields/SmartInput.tsx:139]: It looked bad!!! and inside the input field

File: apps/web/src/modules/create-proposal/constants/transactionType.tsx (hunk 3 above)
Comment (wtfsayo) [reply apps/web/src/modules/create-proposal/constants/transactionType.tsx:45]: Yeah as per figma ig

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowUtils.ts (hunk 4 above)
Comment (wtfsayo) [reply apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowUtils.ts:7]: I am trying to avoid installing any package!

**Review by dan13ram — COMMENTED** [review]
Looks good to me. Just some minor comments

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/Escrow.tsx (hunk 5 above)
Comment (wtfsayo) [reply apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/Escrow.tsx]: `handleEscrowTransaction` ? sure can do

File: apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts (hunk 6 above)
Comment (wtfsayo) [reply apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts]: could do!

**Pull request merged**